src/analysis/bounded_variation.lean

```diff
@@ -207,6 +206,48 @@ begin
   simp [u, edist_comm],
 end
 
+lemma lower_continuous_aux {ι : Type*} {F : ι → α → E} {p : filter ι}
+  {f : α → E} {s : set α} (Ffs : ∀ x ∈ s, tendsto (λ i, F i x) p (𝓝 (f x)))
+  {v : ℝ≥0∞} (hv : v < evariation_on f s) : (∀ᶠ (n : ι) in p, v < evariation_on (F n) s) :=
+begin
+  obtain ⟨⟨n, ⟨u, um, us⟩⟩, hlt⟩ :
+    ∃ (p : ℕ × {u : ℕ → α // monotone u ∧ ∀ i, u i ∈ s}),
+      v < ∑ i in finset.range p.1, edist (f ((p.2 : ℕ → α) (i+1))) (f ((p.2 : ℕ → α) i)) :=
+    lt_supr_iff.mp hv,
```

Comment:
Not sure why we want the long type when using `obtain`; we don't usually do this with `intro`/`rintro` right?

Reply:
`obtain` was introduced precisely in order to make proofs more readable because one could write down the type.

Reply:
I think you can also do `⟨⟨(n : ℕ), (u : ℕ → α), um, us⟩, hlt⟩` with `rintro` if you want to emphasize types of certain variables.